lms/views/lti/basic_launch.py

```diff
@@ -82,33 +82,37 @@ def unconfigured_launch(self):
 
         This happens if we cannot resolve a document URL for any reason.
         """
+        return self._configure_js_for_file_picker()
 
-        if not self.request.has_permission(Permissions.LTI_CONFIGURE_ASSIGNMENT):
-            # Looks like the user is not an instructor, so show an error page
-
-            # https://docs.pylonsproject.org/projects/pyramid/en/latest/narr
-            # /renderers.html?highlight=override_renderer#overriding-a-renderer-at-runtime
-            self.request.override_renderer = "lms:templates/lti/basic_launch/unconfigured_launch_not_authorized.html.jinja2"
-            return {}
-
-        form_fields = {
-            param: value
-            for param, value in self.request.lti_params.items()
-            # Don't send over auth related params. We'll use our own
-            # authorization header
-            if param
-            not in ["oauth_nonce", "oauth_timestamp", "oauth_signature", "id_token"]
-        }
-
-        form_fields["authorization"] = BearerTokenSchema(
-            self.request
-        ).authorization_param(self.request.lti_user)
+    @view_config(
+        route_name="lti.reconfigure",
+        renderer="json",
+    )
```

Comment:
Here I'm using a new view but:

- We could reuse the exact same one but as it uses `has_document_url=False` that's not trivial to change (as other calls will fallback there.

- We might be able to use the same view with another viewconfig maybe, one that checks that we are getting json and `has_document=True`

I reckon this would be easy to change in any case.